### PR TITLE
Remove dev mode and make non-prod CLI mode hot reload

### PR DESCRIPTION
**Pull request by wwwillchen**
- The dev mode wasn't actually working (after I changed our hot reload mechanism to use polling) and I personally haven't used it in months.
- The main benefits of dev mode was: 1) faster client builds, which is fixed for the regular CLI in #316 and 2) server reloading, which I've done here.
- By reducing the number of modes that Mesop runs in, it'll reduce the overall maintenance burden and avoid issues where it works in one mode but not the other modes.

### CODE REVIEW

**Review by richard-to — APPROVED** [review]

File: docs/internal/modes.md [hunk 1]
```diff
@@ -1,28 +1,18 @@
 # Modes
 
-There are several modes that you can run Mesop in.
+There are two modes that you can run Mesop in.
 
-## Uncompiled mode
+## Editor mode (aka debug mode)
```
[the remark this answers is not in the document]
Comment (wwwillchen) [reply docs/internal/modes.md:5]: good call

File: docs/internal/modes.md [hunk 2]
```diff
@@ -1,28 +1,18 @@
...
-## Debug mode
-
-Recommended for developers using Mesop when they are developing the apps locally. This provides good error messages and, in the future, will provide hot reloading.
+Recommended for developers using Mesop when they are developing the apps locally. This provides good error messages and hot reloading.
 
 - **How to run:** `$ bazel run //mesop/cli -- --path=mesop/mesop/example_index.py --ci`
 - Angular should run in dev mode.
 - Developer Tools are available.
 
 ## Prod mode
```
[the remark this answers is not in the document]
Comment (wwwillchen) [reply docs/internal/modes.md:13]: added command

File: docs/internal/modes.md [hunk 3]
```diff
@@ -1,28 +1,18 @@
...
-Only used for developing the Mesop framework itself. Developers _using_ Mesop shouldn't need to use this mode or be concerned about it.
-
-- **How to run:** `$ scripts/run_py_dev.sh` and `$ scripts/run_web_dev.sh`
-- Provides fast compile times, but requires two concurrent Bazel commands.
-- Angular is run in dev mode.
-
-## Debug mode
-
-Recommended for developers using Mesop when they are developing the apps locally. This provides good error messages and, in the future, will provide hot reloading.
+Recommended for developers using Mesop when they are developing the apps locally. This provides good error messages and hot reloading.
```
[the remark this answers is not in the document]
Comment (wwwillchen) [reply docs/internal/modes.md:7]: added

**Comment by wwwillchen** [comment]
Thanks for the review.

**Pull request merged**